qt/scientific_interfaces/General/StepScan.cpp

```diff
@@ -544,9 +546,13 @@ bool StepScan::runStepScanAlgLive(std::string stepScanProperties) {
   IAlgorithm_const_sptr oldMonitorLiveData =
       m_uiForm.mWRunFiles->stopLiveAlgorithm();
 
-  stepScanProperties.erase(0, stepScanProperties.find_first_of('(') + 1);
-  stepScanProperties.erase(stepScanProperties.find_last_of(')'));
-  std::replace(stepScanProperties.begin(), stepScanProperties.end(), ',', ';');
+  Json::Value root;
+  Json::Reader reader;
+  bool parsingSuccessful = reader.parse(stepScanProperties.c_str(), root);
+  if (!parsingSuccessful) {
+    throw std::runtime_error("Parsing parameters failed for StepScan.");
+  }
+  std::string ssp = root.get("properties", "").toStyledString();
```

Comment:
Is getting the default value always an error? Should we check?
Similarly, would it be clearer to use operator[](const std::string &) and check for a null value?